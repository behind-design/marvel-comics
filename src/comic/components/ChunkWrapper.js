import React from 'react';
import chunk from 'lodash.chunk';


const ChunkWrapper = ({ name, items, expand }) => (
  <div className={`${name}-wrapper`}>
    {
      Array.isArray(items) && items.length
        ? chunk(items, 3).map((chunk, idx) => (
          <div key={`${name}-chunk-${idx}`}>
            {chunk.map(item => <p key={expand(item)}>{expand(item)}</p>)}
          </div>
        ))
        : <p>No {name} available.</p>
    }
  </div>
);

export default ChunkWrapper;
