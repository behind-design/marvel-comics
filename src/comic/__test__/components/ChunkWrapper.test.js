import React from 'react';
import ReactDOM from 'react-dom';
import { mount, render, shallow } from 'enzyme';

import ChunkWrapper from '../../components/ChunkWrapper';


describe('ChunkWrapper', () => {
  it('renders without crashing', () => {
    const props = {
      name: 'creators',
      items: null,
      expand: jest.fn(),
    }
    render(
      <ChunkWrapper {...props} />
    );
  });

  it('renders all the correct properties', () => {
    const props = {
      name: 'Yeah',
      items: [{ name: 'Test' }],
      expand: (item) => item.name,
    }
    const wrapper = shallow(
      <ChunkWrapper {...props} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
