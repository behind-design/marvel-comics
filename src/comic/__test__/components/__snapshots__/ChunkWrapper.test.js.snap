// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChunkWrapper renders all the correct properties 1`] = `
<div
  className="Yeah-wrapper"
>
  <div
    key="Yeah-chunk-0"
  >
    <p
      key="Test"
    >
      Test
    </p>
  </div>
</div>
`;
